code/controllers/subsystem/codex.dm

```diff
@@ -168,3 +241,120 @@ SUBSYSTEM_DEF(codex)
 		if(entry)
 			present_codex_entry(showing_mob, entry)
 			return TRUE
+
+#define CODEX_SEARCH_INDEX_FILE "codex.db"
+#define CODEX_SERIAL_ALWAYS_VALID "I_DOWNLOADED_A_ZIP_INSTEAD_OF_USING_GIT"
+/// Prepare the search database.
+/datum/controller/subsystem/codex/proc/prepare_search_database(drop_existing = FALSE)
+	if(GLOB.is_debug_server && !FORCE_CODEX_DATABASE)
+		to_chat(world, span_debug("Codex: Debug server detected. DB operation disabled."))
+		return
+	if(drop_existing)
```

Comment:
is there ever a case where you WOULDNT wnat to delete the existing database? wouldnt it just get overridden due to sharing the same name anyway?

Reply:
> is there ever a case where you WOULDNT want to delete the existing database?

This proc is used to both load the cached database, and re-generate it.

> wouldnt it just get overridden due to sharing the same name anyway?

No, that would result in us *opening* the database.